inference-engine/include/ie_blob.h

```diff
@@ -577,13 +577,19 @@ class TBlob : public MemoryBlob {
     /**
      *@brief Virtual destructor.
      */
-#ifdef __clang__
-    virtual ~TBlob();
+#ifdef __SYCL_COMPILER_VERSION
+    virtual ~TBlob() {
+        free();
+    }
 #else
+#  ifdef __clang__
+    virtual ~TBlob();
+#  else
     virtual ~TBlob() {
         free();
     }
-#endif  // __clang__
+#  endif  // __clang__
+#endif  // __SYCL_COMPILER_VERSION
```

Comment:
```cpp
#if defined(__clang__) && !defined(__SYCL_COMPILER_VERSION)
    virtual ~TBlob();
#else
    virtual ~TBlob() {
        free();
    }
#endif
```

Reply:
I have reflected your suggestion and pushed the commit again. Thank you.